src/main/java/uk/gov/hmcts/reform/blobrouter/data/events/EnvelopeEventRepository.java

```diff
@@ -28,6 +28,14 @@ public EnvelopeEventRepository(NamedParameterJdbcTemplate jdbcTemplate, Envelope
         );
     }
 
+    public List<EnvelopeEvent> findForEnvelopes(List<UUID> envelopeIds) {
+        return jdbcTemplate.query(
+            "SELECT * FROM envelope_events WHERE envelope_id in (:envelopeIds) ORDER BY id DESC",
```

Comment:
probably should make sure its not empty list. maybe even use specific class for that

Reply:
Good point! I'll update

Reply:
Is it needed? In blob-router, unlike in bulk-scan-processor, envelopes always exist.

Reply:
but there may be not any envelopes matching for the search criteria (file_name, container and date)

Reply:
Aaa, input list. Good point. Probably best to do the check in the service.

Reply:
Updated service method to check if envelopes list is empty